### ISSUE TITLE
customizing helm-locate-recursive-dirs-command

### ISSUE THREAD

**Issue by maindoor** [emacs-helm/helm#1709]
## Expected behavior
I changed the variable helm-locate-recursive-dirs-command 
that is used during helm-find-files from
locate -i -e -A --regex '^%s' '%s.*$' 
to 
find . -iregex '.*%s.*'

When I run the command manually from a command line I am able to get
a list of entries that I am searching for. But when I use helm-find-files
the list is empty. and it immediately prompts me for a pattern.

## Steps to reproduce (recipe)

I have outlined this in Expected behavior

## Describe versions of helm, emacs, operating system etc.
helm version 20170216.738 installed from melpa
emacs version 24.5.1
Operating System Kubuntu 16.04

**Comment by thierryvolpiatto** [comment]
maindoor <notifications@github.com> writes:

> Expected behavior
>
> I changed the variable helm-locate-recursive-dirs-command
> that is used during helm-find-files from
> locate -i -e -A --regex '^%s' '%s.$'
> to
> find . -iregex '.%s.*'

First of all, find is too slow to work reliabily from e.g $HOME, you
should use a tool using a db.

The two format specs are mandatory (I have just updated docstring).
Basically one is specifying which directory you want to start your
command from and the second is specifying the basename of pattern.

The "." given as first arg of find would search in default-directory
only (i.e the directory from where you started HFF session) which is not
what you want.

-- 
Thierry

**Comment by maindoor** [comment]
Agree find is too slow, but I have several use cases where I want to use find. Eg. sshfs mounted directories, temporary log directories.
Also agree that searching from "." may not be what I want, but I can fix that later once helm is able to work from "." Can you give me a find string that helm can work with ? I don't have access to my computer now, I will try it out when I get to work on Monday.

**Comment by thierryvolpiatto** [comment]
maindoor <notifications@github.com> writes:

> Agree find is too slow, but I have several use cases where I want to
> use find. Eg. sshfs mounted directories, temporary log directories.
> Also agree that searching from "." may not be what I want, but I can
> fix that later once helm is able to work from "." Can you give me a
> find string that helm can work with ?

No, this have been implemented for use with locate, find is inefficient
for this and I will not make this working with it.
What you should try instead is using find with `C-c /` command from
helm-find-files.

--
Thierry

**Comment by maindoor** [comment]
It is just a way to narrow down options using find instead of locate. The option is there, just that I need to make it work. The work-flow becomes convoluted because of this. For example if I have a/b/c/d/f/some.c and wanted to grep something inside the dir where some.c is present.
If "a" is a sshfs mounted dir and if I just downloaded the contents of a for a short while (not yet updatedb'ed), then to grep for something inside f. Currently I am at "a" then 'C-c /' search for some.c then do a C-x C-f, then a C-l then continue with find-files again. 
If find was working properly I could just do a "..some.c/" and then a C-l and continue from there. 

IMHO...When you provide a functionality, it must work with all inputs, brushing off saying it is inefficient when I have provided a valid use cause is just alienating supporters. and I really wanted to avoid making a github workspace just for this one small functionality.
